### PR TITLE
fixing jog_arm segfault

### PR DESCRIPTION
**Pull request by mlautman**
### Description
Fixes a segfault issue with jog_arm where the fix bounds method tries to access invalid indices inthe velocity vector. Not sure if this is the best way to fix this. @AndyZe

### CODE REVIEW

File: moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp [hunk 1]
```diff
@@ -568,13 +568,18 @@ void JogCalcs::enforceJointVelocityLimits(Eigen::VectorXd& calculated_joint_chan
...
   for (auto joint : joint_model_group_->getJointModels())
   {
     if (!kinematic_state_->satisfiesVelocityBounds(joint))
     {
       ROS_WARN_STREAM_THROTTLE_NAMED(2, LOGNAME, ros::this_node::getName() << " " << joint->getName() << " "
                                                                            << " close to a "
                                                                               " velocity limit. Enforcing limit.");
+
       kinematic_state_->enforceVelocityBounds(joint);
       for (std::size_t c = 0; c < num_joints_; ++c)
```
Comment (AndyZe) [inline moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp]: for (std::size_t c = 0; c < new_jt_traj.points[0].velocities.size(); ++c)

File: moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp [hunk 2]
```diff
@@ -568,13 +568,18 @@ void JogCalcs::enforceJointVelocityLimits(Eigen::VectorXd& calculated_joint_chan
 bool JogCalcs::checkIfJointsWithinURDFBounds(trajectory_msgs::JointTrajectory& new_jt_traj)
 {
   bool halting = false;
+
+  if (new_jt_traj.points[0].velocities.size() < num_joints_)
+    new_jt_traj.points[0].velocities.resize(num_joints_);
```
Comment (AndyZe) [inline moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp]: Would prefer to delete this and move the check to line 584. I think that's a cleaner way

Comment (mlautman) [reply moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp]: agreed. I just made local changes to do exactly that.

File: moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp [hunk 3]
```diff
@@ -588,15 +590,14 @@ bool JogCalcs::checkIfJointsWithinURDFBounds(trajectory_msgs::JointTrajectory& n
 
     // Halt if we're past a joint margin and joint velocity is moving even farther past
     double joint_angle = 0;
-    for (std::size_t c = 0; c < num_joints_; ++c)
+    for (std::size_t c = 0; c < joint_angle = original_jt_state_.size(); ++c)
```
Comment (AndyZe) [inline moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp]: Oops, typo here. If you don't mind, I'll make the change

Comment (mlautman) [reply moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp]: fixed

Comment (mlautman) [reply moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp]: That's what I get for managing the commit separately from my working environment :P

File: moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp [hunk 4]
```diff
@@ -568,15 +568,17 @@ void JogCalcs::enforceJointVelocityLimits(Eigen::VectorXd& calculated_joint_chan
...
   {
     if (!kinematic_state_->satisfiesVelocityBounds(joint))
     {
       ROS_WARN_STREAM_THROTTLE_NAMED(2, LOGNAME, ros::this_node::getName() << " " << joint->getName() << " "
                                                                            << " close to a "
                                                                               " velocity limit. Enforcing limit.");
+
       kinematic_state_->enforceVelocityBounds(joint);
-      for (std::size_t c = 0; c < num_joints_; ++c)
+      for (std::size_t c = 0; c < new_jt_traj.points[0].velocities.size(); ++c)
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp]: Small simplification / homogenization:
```suggestion
      for (std::size_t c = 0; c < new_jt_traj.joint_names.size(); ++c)
```

**Pull request merged**